compiler/src/test/java/com/squareup/anvil/compiler/dagger/ProvidesMethodFactoryGeneratorTest.kt

```diff
@@ -3357,6 +3357,44 @@ public final class DaggerModule1_ProvideFunctionFactory implements Factory<Set<S
     }
   }
 
+  @Test
+  fun `a provides method with starting with 'is' but not as a word is supported`() {
+    compile(
+      """
+      package com.squareup.test
+      
+      import dagger.Module
+      import dagger.Provides
+      import javax.inject.Singleton
+      
+      @Module
+      class DaggerModule1 {
+        @get:Provides
+        val issues: List<String> = listOf("a")
```

Comment:
Could you also add a test for the _ case? Like `is_valid_cache`. It looks like we don't currently have a test for that case in the preexisting tests